pulp_rpm/app/tasks/publishing.py

```diff
@@ -117,18 +117,46 @@ def publish_artifacts(self, content, prefix=""):
             content__pulp_type=Package.get_pulp_type()
         )
 
+        paths = set()
+        duplicated_paths = []
         for content_artifact in contentartifact_qs.values("pk", "relative_path").iterator():
             relative_path = content_artifact["relative_path"]
             relative_path = os.path.join(
                 prefix, PACKAGES_DIRECTORY, relative_path.lower()[0], relative_path
             )
+            #
+            # Some Suboptimal Repos have the 'same' artifact living in multiple places.
```

Comment:
Just to check my own understanding, the problem is that we store only the filename in "relative_path" (because storing "packages/f/filename.rpm" is a bad idea), but the filename could be present in two different places in the directory structure independently, so Pulp says "there's a conflict here" but in reality there is not.

Correct?

Reply:
Yes correct. Good idea, will add more commentary.

Reply:
I'm guessing this will also end up being one of the key pieces of whatever re-architecture we end up doing, because having a full relative-path built into the ContentArtifact seems like an invalid assumption from the beginning.  I think it's discussed in the document already but I'm not sure if the validation details are.

Reply:
I think we really need to step back and think about how "enforcing" Pulp can/should be allowed to be. The Packages/n/name.rpm layout is (clearly) not universal, "multiple checksums for the same nevra" is allowed, "multiple nevras for one checksum" is **also** allowed, and pulp_rpm's assumptions treat them all as fatal error cases. They probably **should be** Fatal Error Cases - but since, in the rpm-repo-world, "anything not expressly forbidden by createrepo is Allowed", we're kind of stuck. :(